src/components/structures/UserMenu.tsx

```diff
@@ -292,23 +295,41 @@ export default class UserMenu extends React.Component<IProps, IState> {
                     })}
                 </div>
             )
-        } else if (signupLink) {
+        } else if (signupLink || hostSignupConfig) {
+            let hostSignupAction;
+            if (hostSignupConfig && hostSignupConfig.url) {
+                // If hostSignup.domains is set to a non-empty array, only show
+                // dialog if the user is on the domain or a subdomain.
+                const hostSignupDomains = hostSignupConfig.domains || [];
+                const mxDomain = MatrixClientPeg.get().getDomain();
+                const validDomains = hostSignupDomains.filter(d => (d === mxDomain || mxDomain.endsWith(`.${d}`)));
+                if (!hostSignupDomains || validDomains.length > 0) {
+                    hostSignupAction = <div
+                        className=""
+                        onClick={this.onCloseMenu}
+                    >
+                        <HostSignupAction />
+                    </div>;
+                }
+            }
             topSection = (
-                <div className="mx_UserMenu_contextMenu_header mx_UserMenu_contextMenu_hostingLink">
-                    {_t(
-                        "<a>Upgrade</a> to your own domain", {},
-                        {
-                            a: sub => (
-                                <a
-                                    href={signupLink}
-                                    target="_blank"
-                                    rel="noreferrer noopener"
-                                    tabIndex={-1}
-                                >{sub}</a>
-                            ),
-                        },
-                    )}
-                </div>
+                <>
+                    {signupLink &&
+                        <div className="mx_UserMenu_contextMenu_header mx_UserMenu_contextMenu_hostingLink">
+                            {_t(
+                                "<a>Upgrade</a> to your own domain", {},
+                                {
+                                    a: sub => (
+                                        <a href={signupLink} target="_blank" rel="noreferrer noopener" tabIndex={-1}>
+                                            {sub}
+                                        </a>
+                                    ),
+                                },
+                            )}
+                        </div>
+                    }
+                    {hostSignupAction}
```

Comment:
this looks like it might end up saying

```
Upgrade to your own domain
Get your own Element!
```
as two distinct actions which seems a bit duplicative and strange

Reply:
I was actually under the thinking we would remove the configuration for the old one while adding configuration for the new one, but I realize now, the `hostSignupLink` is shown in the communities panel and somewhere else as well which I can't now remember (profile?). And we would probably want to keep those.

I'll just remove it from here and leave the other instances alone to not risk double instances here.